File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/tsc/projects/l3-component-provider/providerComponent.ts
import * as pulumi from "@pulumi/pulumi";
import * as config from "@pulumi/config";

interface ProviderComponentArgs {
    text: pulumi.Input<string>,
}

export class ProviderComponent extends pulumi.ComponentResource {
    public result: pulumi.Output<string>;
    constructor(name: string, args: ProviderComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:ProviderComponent", name, args, opts);
        const prov = new config.Provider(`${name}-prov`, {name: "my config"}, {
            parent: this,
        });

        const res = new config.Resource(`${name}-res`, {text: args.text}, {
            parent: this,
            provider: prov,
        });

        this.result = res.text;
        this.registerOutputs({
            result: res.text,
        });
    }
}

File: sdk/python/cmd/pulumi-language-python/testdata/setuppy/local/projects/l3-component-provider/providerComponent.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_config as config

class ProviderComponentArgs(TypedDict, total=False):
    text: Input[str]

class ProviderComponent(pulumi.ComponentResource):
    def __init__(self, name: str, args: ProviderComponentArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:ProviderComponent", name, args, opts)

        prov = config.Provider(f"{name}-prov", name="my config",
        opts = pulumi.ResourceOptions(parent=self))

        res = config.Resource(f"{name}-res", text=args["text"],
        opts = pulumi.ResourceOptions(parent=self,
            provider=prov))

        self.result = res.text
        self.register_outputs({
            'result': res.text
        })